### PR TITLE
nimbo: add shell completions

### PR DESCRIPTION
**Pull request by ahldr**
###### Motivation for this change

Requested by nimbo developer: https://github.com/nimbo-sh/nimbo/issues/13

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alex-eyre** [comment]
Result of `nixpkgs-review pr 123165` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nimbo</li>
  </ul>
</details>

**Comment by alex-eyre** [comment]
Result of `nixpkgs-review pr 123165` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nimbo</li>
  </ul>
</details>

**Comment by noreferences** [comment]
Result of `nixpkgs-review pr 123165` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>nimbo</li>
  </ul>
</details>

**Review by jozuas — APPROVED** [review]

File: pkgs/applications/misc/nimbo/default.nix [hunk 1]
```diff
@@ -12,13 +13,20 @@ buildPythonApplication rec {
     rev = "v${version}";
     sha256 = "1fs28s9ynfxrb4rzba6cmik0kl0q0vkpb4zdappsq62jqf960k24";
   };
-
-  propagatedBuildInputs = [ setuptools boto3 awscli requests click pyyaml pydantic ];
+  nativeBuildInputs = [ installShellFiles ];
+  propagatedBuildInputs =
+    [ setuptools boto3 awscli requests click pyyaml pydantic ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/nimbo/default.nix:18]:
```suggestion
  propagatedBuildInputs = [ setuptools boto3 awscli requests click pyyaml pydantic ];
```

also makes the diff more readable.

File: pkgs/applications/misc/nimbo/default.nix [hunk 2]
```diff
@@ -12,13 +13,20 @@ buildPythonApplication rec {
...
 
   # nimbo tests require an AWS instance
   doCheck = false;
   pythonImportsCheck = [ "nimbo" ];
 
+  postInstall = ''
+    installShellCompletion --zsh --name _nimbo <(_NIMBO_COMPLETE=source_zsh $out/bin/nimbo)
+    installShellCompletion --bash --name nimbo.bash <(_NIMBO_COMPLETE=source_bash $out/bin/nimbo)
+    installShellCompletion --fish --name nimbo.fish <(_NIMBO_COMPLETE=source_fish $out/bin/nimbo)
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/nimbo/default.nix:27]:
```suggestion
    installShellCompletion --cmd nimbo \
      --zsh <(_NIMBO_COMPLETE=source_zsh $out/bin/nimbo) \
      --bash <(_NIMBO_COMPLETE=source_bash $out/bin/nimbo) \
      --fish  <(_NIMBO_COMPLETE=source_fish $out/bin/nimbo)
```

**Pull request merged**